pkg/kubelet/kubelet_node_status_test.go

```diff
@@ -3100,16 +3100,18 @@ func TestIsUpdateStatusPeriodExperid(t *testing.T) {
 		expectExpired              bool
 	}{
 		{
+			// Ensure that everything is properly initialized if the node status is not updated before
```

Comment:
this is why we have the detection of changes

https://github.com/kubernetes/kubernetes/blob/ad4d9b2f5119f2bcf742239018a397d3794cb6e0/pkg/kubelet/kubelet_node_status.go#L485

we are trying to be conservative with updates, can't we detect that we need to update instead of blindly update the node after the restart?

Reply:
But the issue is that there is no change. From what I understand, this `updateNode` is making sure the Node status is up to date. If we just restart the kubelet there is no change in the status and no reason to update anything. We need to make sure [setLastObservedNodeAddresses](https://github.com/kubernetes/kubernetes/blob/0b1ef24a9912d1cbbbfe5641bf9cd2cdc91f32bd/pkg/kubelet/kubelet_node_status.go#L747-L751) is called once even if no change has been made. Maybe we can add it to `updateNode`?

Reply:
Just wanted to add that we only ever end up in this situation if the node status has NEVER (since kubelet start) been updated. So it adds a "forced" update on every kubelet (re)start, which is actually the same as the behavior before https://github.com/kubernetes/kubernetes/pull/128640. It would then update the node status after the status report interval no matter if there was a change or not.